omod/src/main/java/org/openmrs/module/appointments/web/controller/RecurringAppointmentsController.java

```diff
@@ -160,4 +164,44 @@ public RecurringAppointmentDefaultResponse getAppointmentByUuid(@RequestParam(va
         }
         return recurringAppointmentMapper.constructResponse(appointment);
     }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/conflicts")
+    @ResponseBody
+    public ResponseEntity<Object> getConflicts(@RequestBody RecurringAppointmentRequest recurringAppointmentRequest) {
+        try {
+            RecurringPattern recurringPattern = recurringAppointmentRequest.getRecurringPattern();
+            Errors errors = new BeanPropertyBindingResult(recurringPattern, "recurringPattern");
+            recurringPatternValidator.validate(recurringPattern, errors);
+            if (!errors.getAllErrors().isEmpty()) {
+                throw new APIException(errors.getAllErrors().get(0).getDefaultMessage());
+            }
+            String clientTimeZone = recurringAppointmentRequest.getTimeZone();
+            Errors timeZoneErrors = new BeanPropertyBindingResult(clientTimeZone, "clientTimeZone");
+            timeZoneValidator.validate(clientTimeZone, timeZoneErrors);
+            if (!timeZoneErrors.getAllErrors().isEmpty()) {
+                throw new APIException(timeZoneErrors.getAllErrors().get(0).getDefaultMessage());
+            }
```

Comment:
error checking and throwing has duplicate code and can be extracted to a method
Eg: `private void checkErrors(Error error) {.........}`

Reply:
Extracted the same, please check.